### ISSUE TITLE
Add ability to download and edit existing Lambda function

### ISSUE THREAD

**Issue by michaelschem** [aws/aws-toolkit-jetbrains#1302]
**Issue**

I can't download and edit lambda functions that already exist in my Amazon account through the AWS explorer.

**Solution**
Add an option to the context menu to download an existing lambda function.

**Comment by asyba** [comment]
@michaelschem the code or only the sam template?

**Comment by michaelschem** [comment]
I'm a bit new to lambdas, but I would think both would be necessary.

**Comment by bnusunny** [comment]
Yes, download both the code and sam template. If a SAM template is already in the project, add the configuration of the function to the exisiting sam template.

**Comment by blevine** [comment]
Why was this issue closed?

**Comment by michaelschem** [comment]
Sorry, it shouldn't have.

**Comment by blevine** [comment]
FWIW, I may have come up with a (non-optimal) workaround.  

1. In IntelliJ IDEA, I copied the source for my node.js lambda into a local file. The lambda icon appeared in the left-hand margin.

1. Create a bare-bones package.json file using npm init.
1. Right-click on lambda icon and choose "Create new...
1. Fill out the information in the form and click Create.

This will go most of the way through creating the lambda, but fail at the end with an error that indicates that the lambda already exists (which makes sense).  But the rest of the creation process will have worked and now your local code is associated with your existing lambda.

I've only just gotten this to work so there may be some more gotchas down the road as I try to work with the existing lambda from IntelliJ.

Regardless, there should be a straightforward way to "import" a lambda that already exists in AWS so that it can be modified/managed from the IDE.

**Comment by bishonbopanna** [comment]
Any update on this ?

**Comment by appelmoesje** [comment]
Any update on this?

**Comment by gmonacho** [comment]
Any update on this?

**Comment by CodePunk-** [comment]
Any update on this?

**Comment by lorenzoriano** [comment]
Any update on this?

**Comment by ConstantineK** [comment]
We'd still very much like to be able to use this in a full featured way!

**Comment by xlinx** [comment]
Any update on this? So far I used vs code plugin to download the back to Intellij....
